dmap.js

```diff
@@ -50,23 +52,20 @@ lib.walk = async (dmap, path) => {
 }
 
 lib.prepareCID = (_cid, lock) => {
-    const cid = Buffer.from(_cid)
-    const numBytes = cid.byteLength
-    if (numBytes >= 64 || numBytes <= 33) throw new Error(`Unsupported CID length.`)
-    const hashStart = numBytes - 32
-    const hash = cid.slice(hashStart)
+    const cid = multiformats.CID.parse(_cid)
+    if (cid.multihash.size != 32) throw new Error(`Unsupported multihash code`)
+    if (cid.version != 1) throw new Error(`Unsupported CID version`)
```

Comment:
dmap should support old cid, this implies high byte needs to be our own prefix byte